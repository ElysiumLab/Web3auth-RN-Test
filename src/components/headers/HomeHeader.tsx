import { View, Text, StyleSheet } from "react-native";

import { RectButton } from "../buttons/Button";
import { COLORS, SIZES } from "../../styles";

export const HomeHeader = ({
  ethAddress,
  redirectUrl,
  onLogin,
  onLogout,
}: any) => {
  return (
    <View style={style.container}>
      <View style={style.logButtonContainer}>
        {ethAddress ? (
          <RectButton
            title="Logout"
            styleTouchable={style.logButtonTouchable}
            styleText={style.logButtonText}
            handlePress={onLogout}
          />
        ) : (
          <RectButton
            title="Login"
            styleTouchable={style.logButtonTouchable}
            styleText={style.logButtonText}
            handlePress={onLogin}
          />
        )}
      </View>

      <View style={style.textContainer}>
        <Text style={style.text}>{`Redirect url: ${redirectUrl}`}</Text>
        <Text style={style.text}>
          {ethAddress ? `Eth Address: ${ethAddress}` : `Eth Address:`}
        </Text>
      </View>
    </View>
  );
};

export default HomeHeader;

const style = StyleSheet.create({
  container: {
    backgroundColor: COLORS.primary,
    height: 100,
  },

  textContainer: {
    position: "absolute",
    bottom: 10,
  },

  text: {
    color: COLORS.white,
    fontSize: SIZES.small,
  },

  logButtonContainer: {
    width: 60,
    right: 10,
    top: 10,
    position: "absolute",
  },

  logButtonTouchable: {
    backgroundColor: COLORS.blue,
    minHeight: 30,
  },
  logButtonText: {
    color: COLORS.white,
    fontSize: SIZES.medium,
  },
});
