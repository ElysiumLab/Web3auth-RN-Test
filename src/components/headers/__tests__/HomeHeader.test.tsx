import React from "react";
import { render, fireEvent, screen } from "@testing-library/react-native";

import { HomeHeader } from "../HomeHeader";

describe("<HomeHeader />", () => {
  it("renders correctly", () => {
    const tree = render(<HomeHeader />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("press Login correctly", async () => {
    const spyLogin = jest.fn();

    render(<HomeHeader onLogin={spyLogin} />);

    const loginButton = await screen.findByText("Login");
    fireEvent.press(loginButton);

    expect(spyLogin).toBeCalled();
  });

  it("press Logout correctly", async () => {
    const spyLogout = jest.fn();

    render(<HomeHeader ethAddress="test" onLogout={spyLogout} />);

    const loginButton = await screen.findByText("Logout");
    fireEvent.press(loginButton);

    expect(spyLogout).toBeCalled();
  });
});
