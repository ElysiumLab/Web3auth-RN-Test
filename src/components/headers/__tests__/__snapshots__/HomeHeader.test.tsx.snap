// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeHeader /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#001F2D",
      "height": 100,
    }
  }
>
  <View
    style={
      Object {
        "position": "absolute",
        "right": 10,
        "top": 10,
        "width": 60,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#0000FF",
          "justifyContent": "center",
          "minHeight": 30,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "#FFFFFF",
              "fontSize": 16,
            },
          ]
        }
      >
        Login
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 10,
        "position": "absolute",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 12,
        }
      }
    >
      Redirect url: undefined
    </Text>
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 12,
        }
      }
    >
      Eth Address:
    </Text>
  </View>
</View>
`;
